pkg/controller/controller_utils.go

```diff
@@ -97,7 +97,7 @@ var UpdateLabelBackoff = wait.Backoff{
 
 var (
 	KeyFunc           = cache.DeletionHandlingMetaNamespaceKeyFunc
-	podPhaseToOrdinal = map[v1.PodPhase]int{v1.PodPending: 0, v1.PodUnknown: 1, v1.PodRunning: 2}
+	podPhaseToOrdinal = map[v1.PodPhase]int{v1.PodPending: 0, v1.PodRunning: 1}
```

Comment:
dropping this means any pods with phase PodUnknown will be treated the same as PodPending (zero value for missing key). Is that intentional?

Reply:
same comment applies to other sort maps that are dropping the PodUnknown key

Reply:
The idea is that there will never be pods with the status `PodUnknown` as the only codepath setting it was removed.

The original PR #88073 had a valid comment that for proper deprecation, it shouldn't be used anywhere. Thus - removing it from sorting functions.

Reply:
Now the only mention of `podUnknown` is in pkg/apis/core/types.go

Reply:
just because nothing in-tree encounters the value, it doesn't mean the value cannot be persisted into the API, right? if so, the sorting code should continue to handle it

Reply:
I'm happy to return the sorting code back. But before doing it I'm struggling to understand how this may actually happen. There are no code that will set the phase to podUnknown. So for pod to be persisted with this status is needs to survive thru many kubelet upgrades. Moreover the status podUnknown was caused by connectivity issue and likely will be restored thru updates.

Or you are saying that something else outside the kubelet can set this phase for the pod and it will not be accounted in sorting?

Reply:
> Or you are saying that something else outside the kubelet can set this phase for the pod and it will not be accounted in sorting?

This.